tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts

```diff
@@ -809,187 +809,275 @@ describe('image card', () => {
+            start: {step: 19},
+            end: null,
+          });
+          const timeSeries = [
+            {wallTime: 100, imageId: 'ImageId1', step: 10},
+            {wallTime: 101, imageId: 'ImageId2', step: 20},
+            {wallTime: 102, imageId: 'ImageId3', step: 30},
+            {wallTime: 103, imageId: 'ImageId4', step: 40},
+          ];
+          provideMockCardSeriesData(
+            selectSpy,
+            PluginType.IMAGES,
+            'card1',
+            null /* metadataOverride */,
+            timeSeries,
+            2 /* stepIndex */
+          );
+          const fixture = createImageCardContainer('card1');
+          fixture.detectChanges();
+          let slider = fixture.debugElement.query(By.css('mat-slider'));
+          expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('1');
+        });
 
-    it('does not move slider thumb to selected time on range selection', () => {
-      store.overrideSelector(selectors.getMetricsSelectedTime, {
-        start: {step: 15},
-        end: {step: 55},
-      });
-      const timeSeries = [
-        {wallTime: 100, imageId: 'ImageId1', step: 10},
-        {wallTime: 101, imageId: 'ImageId2', step: 20},
-        {wallTime: 102, imageId: 'ImageId3', step: 30},
-        {wallTime: 103, imageId: 'ImageId4', step: 40},
-      ];
-      provideMockCardSeriesData(
-        selectSpy,
-        PluginType.IMAGES,
-        'card1',
-        null /* metadataOverride */,
-        timeSeries,
-        2 /* stepIndex */
-      );
-      const fixture = createImageCardContainer('card1');
-      fixture.detectChanges();
+        it('does not move slider thumb to larger closest step when it is clipped', () => {
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 9},
+            end: null,
+          });
+          const timeSeries = [
+            {wallTime: 100, imageId: 'ImageId1', step: 10},
+            {wallTime: 101, imageId: 'ImageId2', step: 20},
+            {wallTime: 102, imageId: 'ImageId3', step: 30},
+            {wallTime: 103, imageId: 'ImageId4', step: 40},
+          ];
+          provideMockCardSeriesData(
+            selectSpy,
+            PluginType.IMAGES,
+            'card1',
+            null /* metadataOverride */,
+            timeSeries,
+            2 /* stepIndex */
+          );
+          const fixture = createImageCardContainer('card1');
+          fixture.detectChanges();
+          let slider = fixture.debugElement.query(By.css('mat-slider'));
+          expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
+        });
 
-      let slider = fixture.debugElement.query(By.css('mat-slider'));
-      expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
+        it('does not move slider thumb to smaller closest step when it is clipped', () => {
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 41},
```

Comment:
Can you add an explanation of why this is clipped (I can't figure it out)? (And perhaps the additional information of "if this were not clipped, 41 is close enough to 40 so would otherwise choose step 3)